text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,284 @@
+    detectors: [env, docker]
+    timeout: 2s
+    override: false
+```
+
+The future entity-based detector would look almost exactly the same, but interact with the entity model of resource:
+
+```yaml
+processor:
+  entityresourcedetection:
+     # Order determines override behavior
+     detectors: [env, docker]
+     # False means only append if entity doesn't already exist.
+     override: false 
+```
+
+The list of detectors is given in priority order (first wins, in event of a tie, outside of override configuration). The processor may need to be updated to allow the override flag to apply to each individual detector.
+
+## Datamodel Changes
+
+Given our desired design and algorithms for detecting, merging and manipulating Entities, we need the ability to denote how entity and resource relate. These changes must not break existing usage of Resource, therefore:
+
+- The Entity model must be *layered on top of* the Resource model.  A system does not need to ineract with entities for correct behavior.
+- Existing key usage of Resource must remain when using Entities, specifically navigationality (see: [OpenTelemetry Resources: Principles and Characteristics](https://docs.google.com/document/d/1Xd1JP7eNhRpdz1RIBLeA1_4UYPRJaouloAYqldCeNSc/edit))
+- Downstream components should be able to engage with the Entity model in Resource.
+
+The following changes are made:
+
+### Resource
+
+| Field | Type | Description | Changes |
+| ----- | ---- | ----------- | ------- |
+| schema_url | string | The Schema URL, if known. This is the identifier of the Schema that the resource data  is recorded in.  This field is deprecated and should no longer be used. | Will be deprecated |
+| dropped_attributes_count |  integer | dropped_attributes_count is the number of dropped attributes. If the value is 0, then no attributes were dropped. | Unchanged |
+| attributes | repeated KeyValue | Set of attributes that describe the resource.<br/><br/>Attribute keys MUST be unique (it is not allowed to have more than one attribute with the same key).| Unchanged |
+| entities | repeated ResourceEntityRef | Set of entities that participate in this Resource. | Added |
+
+The DataModel would ensure that attributes in Resource are produced from both the identifying and descriptive attributes of Entity.  This does not mean the protocol needs to transmit duplicate data, that design is TBD.
+
+### ResourceEntityRef
+
+The entityref data model, would have the following changes from the original [entity OTEP](https://github.com/open-telemetry/oteps/blob/main/text/entities/0256-entities-data-model.md) to denote references within Resource:
+
+| Field | Type | Description | Changes |
+| ----- | ---- | ----------- | ------- |
+| schema_url | string | The Schema URL, if known. This is the identifier of the Schema that the entity data  is recorded in. To learn more about Schema URL ([see docs](https://opentelemetry.io/docs/specs/otel/schemas/#schema-url)) | added |
```

Comment:
Nit: can we capitalize field names to match the style used in previous data model documents (e.g. https://github.com/open-telemetry/oteps/blob/main/text/entities/0256-entities-data-model.md#data-model)?

Reply:
I have them matching the proto, i.e. I expect these to be the proto changes we define.  I'm happy to change, but want to make sure we understand proposed proto changes.

Reply:
OK, I am fine with either approach as long as it is consistent. If we use snake_case for entities, I think we should go and update logs and entities docs to use that too.